watchtower/wtmock/client_db.go

```diff
@@ -172,25 +172,26 @@ func (m *ClientDB) CommitUpdate(id *wtdb.SessionID, seqNum uint16,
 	}
 
 	// Check if an update has already been committed for this state.
-	dbUpdate, ok := session.CommittedUpdates[seqNum]
-	if ok {
-		// If the breach hint matches, we'll just return the last
-		// applied value so the client can retransmit.
-		if dbUpdate.Hint == update.Hint {
-			return session.TowerLastApplied, nil
+	for _, dbUpdate := range session.CommittedUpdates {
+		if dbUpdate.SeqNum == update.SeqNum {
```

Comment:
I guess this is the downside then to keeping it in a sorted slice instead of a map. Lookup could probably be faster with some kind of hierarchical split search algo, but likely a premature optimization.

Reply:
Indeed, the map translates better as far as a model for the on-disk structure. But in fact, this is pretty much what bbolt is doing under the hood :P